blog/2021-05-19-monero-transaction.md

```diff
@@ -0,0 +1,280 @@
+---
+title: "How to build a Monero transaction"
+author: lucas
+author_url: https://github.com/luckysori
+author_image_url: https://avatars3.githubusercontent.com/u/9418575
+tags: [monero]
+---
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+Here's another post on one of our favourite privacy coins.
+This time we will take a look at the technical details of Monero transactions.
+Most of you will know that Monero transactions are confidential in terms of amounts and ownership of outputs, but here we'll explain what makes this work.
+
+Our motivation to take a deep dive into the inner works of Monero transactions stems from a desire to be able to construct and sign them manually.
+There are two ways in which we would benefit from this:
+
+1. Our [swap tool](https://github.com/comit-network/xmr-btc-swap/) based on ["Bitcoin-Monero Cross-chain Atomic Swap"](https://eprint.iacr.org/2020/1126.pdf) currently relies on `monero-wallet-rpc` to build the spend transactions, which prevents us from running the swap in the browser.
+2. Our ["Monero moves first" atomic swap protocol](https://arxiv.org/pdf/2101.12332.pdf) uses CLSAG adaptor signatures, which can only be constructed with direct control over how transactions are built and signed.
+
+## Anatomy of a Monero transaction
+
+In this section we will focus on Monero transactions of type `5` (`CLSAG`), which is the new transaction type since fork version 13.
+Monero changes at a relatively high speed compared to other cryptocurrencies such as Bitcoin, so some of this information may become outdated after the next few planned hard forks.
+
+### Transaction prefix
+
+The transaction prefix is the main part of the transaction, as it includes all the possible inputs and all the outputs.
+
+#### Version
+
+This indicates the mechanism through which output amounts are hidden.
+Nowadays this value is set to `2`, which corresponds to Ring Confidential Transactions (RingCT).
+
+#### Unlock time aka `unlock_time`
+
+It is easy to confuse this field with Bitcoin's `nLocktime`, but there are subtle differences.
+Where `nLocktime` prevents a Bitcoin transaction from being included in a block before a certain time, the value of `unlock_time` just determines how long all recipients of the outputs of a Monero transaction will have to wait until they can spend them.
+
+#### Inputs aka `vin`
+
+##### Amount
+
+This is a legacy field and is always set to `0`.
+Amount confidentiality used to be opt-in, but it is mandatory as of fork version 6.
+
+##### Key offsets
+
+Where the coins may come from for a particular transaction input.
+The key word is _may_.
+We include a set of 11 possible outputs for each input, but only one of them is the actual source.
+This feature ensures that an uninvolved onlooker cannot definitively draw relationships between transactions.
```

Comment:
I was trying to avoid stating that transactions cannot be linked at all, because I think there might be heuristics which can get you there. I'll reword it for clarity.